src/Cache/Mutator/ResponseMutator.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+namespace Http\Client\Common\Plugin\Cache\Mutator;
+
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\ResponseInterface;
+
+/**
+ * An interface for a mutator of a possibly cached response.
+ *
+ * @author Iain Connor <iain.connor@priceline.com>
+ */
+interface ResponseMutator
```

Comment:
i am not happy with this name. every plugin has the possibility to change a response. this thing is specifically about the cache. but unfortunately i don't have a great name either. `CacheListener` is not too good, but would be more specific to what this does. can we be even more specific?

the class doc could be "XXX is called by the cache plugin with information on the cache status to change information on the response based on whether it was a cache hit or a miss."

Reply:
I can't think of anything significantly better. Will rename this for the time being and see how it feels.